### ISSUE TITLE
Sporadic OpenSSL decryption fails for HttpClient responses

### ISSUE THREAD

**Issue by JoakimJohesan** [dotnet/runtime#73610]
### Description

Hello, we recently began seeing errors reported when doing M2M https requests between our API's. We can see from the application logs that requests are received, and correctly responded to, but when the initiating API attempts to read the response, it gets an decryption exception. The faults occur randomly, and only a very small percentage of our requests fail. (~0,1%). We see no errors on the receiving api end.

Our setup: 
- run on Azure Kubernetes Service 1.22.6. 
- recently upgraded from .Net Core 3.1 LTS to .NET 6.0 LTS. 
- running these Docker images:
  - mcr.microsoft.com/dotnet/aspnet:6.0-alpine
  - mcr.microsoft.com/dotnet/aspnet:6.0
(Checked both and they are both at .Net 6.0.7)
- We are using multiple authentication schemes, each of them trying to authenticate incoming requests. (AddJwtBearer)

We tried using the 6.0.4 dotnet runtime docker image, but still got faults. 

Our issue is very similar to the one reported here: https://github.com/dotnet/runtime/issues/64492

### Reproduction Steps

We have not identified reproduction steps yet. 

### Expected behavior

No errors occuring when using HttpClient to send and receive requests.

### Actual behavior

- Api A sends a request to Api B. 
- B receives request, and responds without errors. 
- A produces either of the following error:

```
---> System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: The decryption operation failed, see inner exception.
 ---> Interop+OpenSsl+SslException: Decrypt failed with OpenSSL error - SSL_ERROR_SSL.
 ---> Interop+Crypto+OpenSslCryptographicException: error:04067072:rsa routines:rsa_ossl_public_decrypt:padding check failed
   --- End of inner exception stack trace ---
   at System.Net.Security.SslStreamPal.DecryptMessage(SafeDeleteSslContext securityContext, Span`1 buffer, Int32& offset, Int32& count)
   --- End of inner exception stack trace ---
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
```

and

```
 System.Net.Http.HttpRequestException: An error occurred while sending the request.
 ---> System.IO.IOException: The decryption operation failed, see inner exception.
 ---> Interop+OpenSsl+SslException: Decrypt failed with OpenSSL error - SSL_ERROR_SSL.
 ---> Interop+Crypto+OpenSslCryptographicException: error:04067084:rsa routines:rsa_ossl_public_decrypt:data too large for modulus
   --- End of inner exception stack trace ---
   at System.Net.Security.SslStreamPal.DecryptMessage(SafeDeleteSslContext securityContext, Span`1 buffer, Int32& offset, Int32& count)
   --- End of inner exception stack trace ---
   at System.Net.Security.SslStream.ReadAsyncInternal[TIOAdapter](TIOAdapter adapter, Memory`1 buffer)
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at System.Net.Http.HttpConnection.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.HttpConnectionPool.SendWithVersionDetectionAndRetryAsync(HttpRequestMessage request, Boolean async, Boolean doRequestAuth, CancellationToken cancellationToken)
   at System.Net.Http.DiagnosticsHandler.SendAsyncCore(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at System.Net.Http.RedirectHandler.SendAsync(HttpRequestMessage request, Boolean async, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at Microsoft.Extensions.Http.Logging.LoggingScopeHttpMessageHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
   at System.Net.Http.HttpClient.<SendAsync>g__Core|83_0(HttpRequestMessage request, HttpCompletionOption completionOption, CancellationTokenSource cts, Boolean disposeCts, CancellationTokenSource pendingRequestsCts, CancellationToken originalCancellationToken)
```

### Regression?

Did not see this problem on .NET Core 3.1 LTS.

### Configuration

.NET 6.0.7 runtime, AKS linux containers, Alpine and Debian distro. 

### Other information

Related issue: https://github.com/dotnet/runtime/issues/64492

**Comment by karelz** [comment]
Triage: We should look into it closer. It is not critical for 7.0 (which forks off next week) as it is not regression against 6.0.

**Comment by rzikm** [comment]
@JoakimJohesan Would you be able to trim the setup to some small(-ish) reproducible example which you could share with us? Without a repro, it is very hard to investigate this.

The issue at https://github.com/dotnet/runtime/issues/64492 was persistent (i.e. SslStream was useless until the process was restarted and the initialization ran successfully), you seem to be having problems only on some requests.

**Comment by bdestercke** [comment]
We have the same issue at one of our clients with the same configuration.
Rolling back to net5.0 seems resolve this, but not exactly a wanted situation since 5.0 is no longer supported.

docker image: mcr.microsoft.com/dotnet/aspnet:6.0-bullseye-slim
Also 6.0.7, taken from the logs of building the image:
"Env": [
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "ASPNETCORE_URLS=http://+:80",/
                "DOTNET_RUNNING_IN_CONTAINER=true",
                "DOTNET_VERSION=6.0.7",
                "ASPNET_VERSION=6.0.7"
            ]
Linux container running in AKS.
kubernetes.azure.com/node-image-version=AKSUbuntu-1804gen2containerd-2022.06.08

Exception message: error:04067072:rsa routines:rsa_ossl_public_decrypt:padding check failed
Logs/stacktrace from app insights:
[{"outerId":"0","message":"An error occurred while sending the request.","type":"System.Net.Http.HttpRequestException","id":"16788047","parsedStack":[{"assembly":"System.Net.Http, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","method":"System.Net.Http.HttpConnection+<SendAsyncCore>d__62.MoveNext","level":0,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":1,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":2,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":3,"line":0},{"assembly":"System.Net.Http, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","method":"System.Net.Http.HttpConnectionPool+<SendWithVersionDetectionAndRetryAsync>d__83.MoveNext","level":4,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":5,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":6,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":7,"line":0},{"assembly":"System.Net.Http, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","method":"System.Net.Http.DiagnosticsHandler+<SendAsyncCore>d__8.MoveNext","level":8,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":9,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":10,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":11,"line":0},{"assembly":"System.Net.Http, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","method":"System.Net.Http.RedirectHandler+<SendAsync>d__4.MoveNext","level":12,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":13,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":14,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":15,"line":0},{"assembly":"Microsoft.Extensions.Http, Version=6.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60","method":"Microsoft.Extensions.Http.Logging.LoggingHttpMessageHandler+<SendAsync>d__5.MoveNext","level":16,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":17,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":18,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":19,"line":0},{"assembly":"Portima.Core.Security.Client, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null","method":"Portima.Core.Security.Client.Http.DelegatingHandlers.TokenCredentialsDelegatingHandler+<SendAsync>d__6.MoveNext","level":20,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":21,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":22,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":23,"line":0},{"assembly":"Microsoft.Extensions.Http.Polly, Version=6.0.7.0, Culture=neutral, PublicKeyToken=adb9793829ddae60","method":"Microsoft.Extensions.Http.PolicyHttpMessageHandler+<SendCoreAsync>d__6.MoveNext","level":24,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":25,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":26,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":27,"line":0},{"assembly":"Polly, Version=7.0.0.0, Culture=neutral, PublicKeyToken=c8a3ffc3f8f825cc","method":"Polly.CircuitBreaker.AsyncCircuitBreakerEngine+<ImplementationAsync>d__0`1.MoveNext","level":28,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":29,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":30,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":31,"line":0},{"assembly":"Polly, Version=7.0.0.0, Culture=neutral, PublicKeyToken=c8a3ffc3f8f825cc","method":"Polly.AsyncPolicy`1+<ExecuteAsync>d__13.MoveNext","level":32,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":33,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":34,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":35,"line":0},{"assembly":"Microsoft.Extensions.Http.Polly, Version=6.0.7.0, Culture=neutral, PublicKeyToken=adb9793829ddae60","method":"Microsoft.Extensions.Http.PolicyHttpMessageHandler+<SendAsync>d__5.MoveNext","level":36,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":37,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":38,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":39,"line":0},{"assembly":"Microsoft.Extensions.Http.Polly, Version=6.0.7.0, Culture=neutral, PublicKeyToken=adb9793829ddae60","method":"Microsoft.Extensions.Http.PolicyHttpMessageHandler+<SendCoreAsync>d__6.MoveNext","level":40,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":41,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":42,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":43,"line":0},{"assembly":"Polly, Version=7.0.0.0, Culture=neutral, PublicKeyToken=c8a3ffc3f8f825cc","method":"Polly.Retry.AsyncRetryEngine+<ImplementationAsync>d__0`1.MoveNext","level":44,"line":0}]},{"outerId":"16788047","message":"The decryption operation failed, see inner exception.","type":"System.IO.IOException","id":"17481261","parsedStack":[{"assembly":"System.Net.Security, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","method":"System.Net.Security.SslStream+<ReadAsyncInternal>d__188`1.MoveNext","level":0,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw","level":1,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess","level":2,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification","level":3,"line":0},{"assembly":"System.Private.CoreLib, Version=6.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e","method":"System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1+ConfiguredValueTaskAwaiter.GetResult","level":4,"line":0},{"assembly":"System.Net.Http, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","method":"System.Net.Http.HttpConnection+<SendAsyncCore>d__62.MoveNext","level":5,"line":0}]},{"outerId":"17481261","message":"Decrypt failed with OpenSSL error - SSL_ERROR_SSL.","type":"Interop+OpenSsl+SslException","id":"62540172","parsedStack":[{"assembly":"System.Net.Security, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a","method":"System.Net.Security.SslStreamPal.DecryptMessage","level":0,"line":0}]},{"outerId":"62540172","message":"error:04067072:rsa routines:rsa_ossl_public_decrypt:padding check failed","type":"Interop+Crypto+OpenSslCryptographicException","id":"38579372"}]

**Comment by karelz** [comment]
@bdestercke can you try to isolate a repro?
Or at least do you have ability to reproduce in some environment where you can experiment? (collect logs, etc.)

@rzikm is there any additional logging that might help us if we do not have standalone repro? (which is IMO unlikely)

**Comment by JoakimJohesan** [comment]
We do not have the time to create a repro at the moment, but are open for enabling logging which can provide insight.

**Comment by rzikm** [comment]
I am not sure, do you know of anything, @wfurt?

**Comment by wfurt** [comment]
I'm not sure if logging would help much since it is possibly inside OpenSSL. As far as the modulus, it seems like there are other posts like https://github.com/openssl/openssl/issues/6841. So it may be specific to particular openssl versions.

It _is_ suspicious that it would start happening after .NET upgrade but we may simply use more functionality from OpenSSL and/or timing and data flow is different.

**Comment by karelz** [comment]
If someone has a repro - I wonder if they could try to update OpenSSL to help us confirm/disproved @wfurt's theory above.

**Comment by JoakimJohesan** [comment]
We are running `OpenSSL 1.1.1n  15 Mar 2022`, what version should we try out?

**Comment by rzikm** [comment]
I would start by downgrading to the same version as on the working image if possible. If it does not reproduce anymore, it is possible that it is not our fault.

**Comment by JoakimJohesan** [comment]
Working image is running `OpenSSL 1.1.1n  15 Mar 2022` as far as i can tell. (Checked with docker) So we are not actively doing anything about our OpenSSL version.

**Comment by JoakimJohesan** [comment]
This looks similar: https://github.com/dotnet/dotnet-docker/issues/3976

I was not able to find OpenSSL on the previous working image, it looks like ssl_client was used there?

**Comment by wfurt** [comment]
do you generate the keys/certificates automatically or is that static/permanent @JoakimJohesan ? 
cc: @bartonjs in case he has more insight into the `rsa_ossl_public_decrypt:data too large for modulus`

**Comment by bartonjs** [comment]
Offhand, something is computing an RSA signature verification, but the signature is N bytes, and the key is M bits, where `M < N * 8`, e.g. the public key is a 2048-bit key and the signature was larger than 256 bytes.

But since I don't overtly see handshaking in the callstack, I'm confused.  Once the session key is established RSA no longer has any role.

There's always the possibility that the message here is bogus.  We cleaned up some error queue processing in https://github.com/dotnet/runtime/commit/45a9548644f34264ce2e1a8a586e8e234d6f61e4#diff-584afd39ea2113de463006322dc18b7358c29f5922ee51ad82bf575ce7278d6c, which is .NET 7.

**Comment by JoakimJohesan** [comment]
> do you generate the keys/certificates automatically or is that static/permanent @JoakimJohesan ? cc: @bartonjs in case he has more insight into the `rsa_ossl_public_decrypt:data too large for modulus`

Certificates are permanent and not managed by the application.

**Comment by wfurt** [comment]
There was similar issue reported internally and upgrade to 7.0 preview fixed it. It may be worth of shot for anybody who can try.

**Comment by rzikm** [comment]
> There's always the possibility that the message here is bogus. We cleaned up some error queue processing in [45a9548#diff-584afd39ea2113de463006322dc18b7358c29f5922ee51ad82bf575ce7278d6c](https://github.com/dotnet/runtime/commit/45a9548644f34264ce2e1a8a586e8e234d6f61e4#diff-584afd39ea2113de463006322dc18b7358c29f5922ee51ad82bf575ce7278d6c), which is .NET 7.

I have cherry-picked https://github.com/dotnet/runtime/pull/65148 on top of 6.0 and the issue does not reproduce anymore, do you think it reasonable to port that change as a fix? @bartonjs, @vcsjones?

**Comment by wfurt** [comment]
yes, this would make sense to me.

**Comment by rzikm** [comment]
Cleaning only in SSL_read and SSL_write works (at least for Release builds where the assert does not trip, and that is what matters). I will put up a PR

**Comment by karelz** [comment]
Thanks for tracking it down @rzikm - I moved the bug into 6.0.x milestone. It does not impact 8.0 (nor 7.0.x).

**Comment by wfurt** [comment]
fixed by #77772 in 6.0.12
